controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -16,48 +16,61 @@
 from controller.tests import utils
 
 
-class AbstractControllerTest(unittest.TestCase):
+class BaseControllerSetUp(unittest.TestCase):
 
     def setUp(self):
         patch("controller.array_action.array_mediator_xiv.XIVArrayMediator._connect").start()
         detect_array_type_patcher = patch("controller.controller_server.csi_controller_server.detect_array_type")
         self.detect_array_type = detect_array_type_patcher.start()
         self.detect_array_type.return_value = "a9k"
         self.addCleanup(detect_array_type_patcher.stop)
+        self.fqdn = "fqdn"
+        self.mediator = XIVArrayMediator("user", "password", self.fqdn)
+        self.mediator.client = Mock()
+
+        self.storage_agent = MagicMock()
+        self.storage_agent.get_mediator.return_value.__enter__.return_value = self.mediator
+
+        self.servicer = ControllerServicer(self.fqdn)
+
+        self.request = Mock()
+
+
+class CommonControllerTest:
 
     @abc.abstractmethod
-    def get_create_object_method(self):
+    def get_tested_method(self):
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_create_object_response_method(self):
+    def get_csi_pb2_response_method(self):

```

Comment:
why `method` and not `type`/`class`?
why do you prefer [using `csi_pb2` name](https://github.com/IBM/ibm-block-csi-driver/pull/212#discussion_r459548226)?

Reply:
because that is the method/object/class origin

Reply:
I don't think we should mix the underlying serialization format (that could theoretically change), with the relevant abstract data structure (i.e. "response class").

Reply:
done